vault/eventbus/bus.go

```diff
@@ -29,10 +29,47 @@ type EventBus struct {
 
 type asyncChanNode struct {
 	// TODO: add bounded deque buffer of *any
-	ch chan any
+	ch chan *logical.EventData
 }
 
-var _ eventlogger.Node = &asyncChanNode{}
+var _ eventlogger.Node = (*asyncChanNode)(nil)
+var _ logical.EventSender = (*EventBus)(nil)
+
+type EventDataWrapper logical.EventData
+
+// ID is a wrapped call to GetEid() for CloudEvents compatibility.
+func (data *EventDataWrapper) ID() string {
```

Comment:
WDYT about implementing the `ID` method directly on the protobuf generated struct instead? It can be in a separate file from the generated file, as long as it's still in the same Go package, we can define extra methods on it, and then you could get rid of a lot of casting, e.g. in `Process`.

Reply:
Ah, good point. I'll do that.